plugins/outputs/bigquery/bigquery.go

```diff
@@ -45,21 +46,21 @@ func (*BigQuery) SampleConfig() string {
 	return sampleConfig
 }
 
-func (s *BigQuery) Connect() error {
-	if s.Project == "" {
-		return fmt.Errorf("Project is a required field for BigQuery output")
-	}
-
+func (s *BigQuery) Init() error {
 	if s.Dataset == "" {
-		return fmt.Errorf("Dataset is a required field for BigQuery output")
+		return errors.New(`"dataset" is required`)
 	}
 
+	s.warnedOnHyphens = make(map[string]bool)
+
+	return nil
+}
+
+func (s *BigQuery) Connect() error {
 	if s.client == nil {
 		return s.setUpDefaultClient()
 	}
 
```

Comment:
Wouldn't it make more sense to disconnect if `s.client` is not `nil`?

Reply:
That is a relic of how the tests are set-up with a mocking http server. (See `setUpTestClient` in bigquery_test.go)